src/RewardsStreamerMP.sol

```diff
@@ -185,25 +262,29 @@ contract RewardsStreamerMP is IStakeManager, TrustedCodehashAccess, ReentrancyGu
         }
     }
 
-    function _updateAccountMP(address accountAddress) internal {
+    function _accruedMP(address accountAddress) internal view returns (uint256) {
         Account storage account = accounts[accountAddress];
-
         if (account.maxMP == 0 || account.stakedBalance == 0) {
-            account.lastMPUpdateTime = block.timestamp;
-            return;
+            return 0;
         }
 
         uint256 timeDiff = block.timestamp - account.lastMPUpdateTime;
         if (timeDiff == 0) {
-            return;
+            return 0;
         }
 
         uint256 accruedMP = (timeDiff * account.stakedBalance * MP_RATE_PER_YEAR) / (365 days * SCALE_FACTOR);
 
         if (account.accountMP + accruedMP > account.maxMP) {
             accruedMP = account.maxMP - account.accountMP;
         }
+        return accruedMP;
+    }
+
+    function _updateAccountMP(address accountAddress) internal {
+        uint256 accruedMP = _accruedMP(accountAddress);
```

Comment:
what do you think about passing the account struct to _accruedMP instead of loading it twice?
I'm not sure about gas, maybe we can test it